packages/postgres-database/src/simcore_postgres_database/migration/versions/a457d9974674_add_optional_parent_node_id.py

```diff
@@ -0,0 +1,47 @@
+"""add optional parent node id
+
+Revision ID: a457d9974674
+Revises: baf0ee1c37dc
+Create Date: 2024-05-27 08:37:31.554118+00:00
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "a457d9974674"
+down_revision = "baf0ee1c37dc"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column(
+        "projects_metadata",
+        sa.Column("parent_project_uuid", sa.String(), nullable=True),
+    )
+    op.add_column(
+        "projects_metadata", sa.Column("parent_node_id", sa.String(), nullable=True)
+    )
+    op.create_foreign_key(
+        "fk_projects_metadata_parent_node_id",
+        "projects_metadata",
+        "projects_nodes",
+        ["parent_project_uuid", "parent_node_id"],
+        ["project_uuid", "node_id"],
+        onupdate="CASCADE",
+        ondelete="SET NULL",
```

Comment:
Q: Why SET NULL and not delete?

Reply:
These are metadata that are set after the fact of creating the "child" study. I am not sure we want to automatically delete the whole metadata information when the parent study is removed.
the project metadata is linked to its direct study.

If we want to have child removal when a parent is removed I think the way we create these relationships should be done explicitely and not via metadata. but I'm open for discussion. I guess then the meta team should be involved as well.